material-stepper/src/main/java/com/stepstone/stepper/adapter/AbstractStepAdapter.java

```diff
@@ -1,19 +1,49 @@
 package com.stepstone.stepper.adapter;
 
 import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.View;
+import android.view.ViewGroup;
 
 import com.stepstone.stepper.Step;
 
-/**
- * Created by leonardo on 18/12/16.
- */
+import java.util.LinkedList;
+import java.util.List;
 
 /**
  * A base adapter class which returns step to use inside of the {@link com.stepstone.stepper.StepperLayout}.
  * This class is intended to be inherited if you need to use {@link com.stepstone.stepper.StepperLayout} without fragments.
- * Otherwise, you should use {@link AbstractStepAdapter}
+ * Otherwise, you should use {@link AbstractFragmentStepAdapter}
  */
-public abstract class AbstractStepAdapter<T extends Step> extends PagerAdapter implements StepAdapter<T> {
+public abstract class AbstractStepAdapter extends PagerAdapter implements StepAdapter {
+
+    private List<Step> pages = new LinkedList<>();
```

Comment:
True, maybe we could use a map here with position-step key-values and remove items from the map in _destroyItem_?